resolversrv/tests/unittests/test_db_query.py

```diff
@@ -327,20 +609,132 @@ def test_link_data(self):
         )
 
 
-    # return 404 for not finding any records
     def test_link_data_error_bibcode(self):
+        """
+        return 404 for not finding any records
+        :return:
+        """
         results = get_records(bibcode='errorbibcode', link_type='DATA')
         response = LinkRequest(bibcode='errorbibcode', link_type='DATA').request_link_type_data(results)
```

Comment:
minor point, but got me wonder -- you are creating an instance, the instance seems to be defined by the initialization arguments (bibcode, link_type)

1) can the api be simpler if you rename `request_link_type_data` to `get_response` ?

if 1) doesn't make sense, is the 'link_type` necessary? could you just do:

2) `LinkRequest(bibcode='errorbibcode').request_type(type='DATA', results)` ?

i guess my main point is: if link_type is 'data', there doesn't need to be a different method for every type

Reply:
Nice catch. Actually LinkRequest basically does not need the params so I changed it to 
        results = get_records(bibcode='errorbibcode', link_type='DATA')
        response = LinkRequest(bibcode='').request_link_type_data(results)
the goal is to make sure request_link_type_data returns error if errored data are in the results.